service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/dataproccluster/startup.sh

```diff
@@ -845,22 +878,40 @@ safe_call("""${RUN_IPYTHON} profile create && echo "c.InteractiveShellApp.extens
 print("hail installed successfully.")
 
 EOF
+fi
 
-  # Fork the following into background process to wait for dataproc to finish
-  # installing optional components including jupyter and start the proxy gateway service.
-  # Then safely install hail.
-  "$(
-    while ! systemctl is-active --quiet ${GOOGLE_DATAPROC_COMPONENT_GATEWAY_SERVICE_NAME}; do
-      sleep 5
-      emit "Waiting for ${GOOGLE_DATAPROC_COMPONENT_GATEWAY_SERVICE_NAME} to start..."
-    done
+# Fork the following into background process to execute after dataproc finishes
+# setting up its optional components.
+#
+# Post dataproc setup tasks:
+# 1. Wait for the dataproc component gateway service to start.
+# 2. Install hail if it has been enabled.
+# 3. Configure jupyter service config
+#    a. Remove Dataproc's GCSContentsManager as we support bucket mounts in local file system.
+#    b. Set jupyter file tree's root directory to the LOGIN_USER's home directory.
+# 4. Restart jupyter service
+#
+"$(
```

Comment:
This is starting to bleed towards something where we should probably write this as a script to /tmp and then run the script in the background, but okay for now.